src/cls/rgw/cls_rgw.cc

```diff
@@ -48,7 +48,7 @@ CLS_NAME(rgw)
 // of a special bucket-index entry for the first byte. Note: although
 // it has no impact, the 2nd, 3rd, or 4th byte of a UTF-8 character
 // may be 0x80.
-#define BI_PREFIX_CHAR 0x80
+constexpr unsigned char BI_PREFIX_CHAR = 0x80;
```

Comment:
```suggestion
constexpr char BI_PREFIX_CHAR = 0x80;
```

Reply:
`std::string` is based on `char` not `unsigned char`

Reply:
@yuvalif  when we use `char`, Jenkins `make-check` generates this error:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/cls/rgw/cls_rgw.cc:83:44: error: result of comparison of constant -128 with expression of type 'unsigned char' is always true [-Werror,-Wtautological-constant-out-of-range-compare]
  return (s.empty() || (unsigned char)s[0] != BI_PREFIX_CHAR);
                       ~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/cls/rgw/cls_rgw.cc:212:45: error: result of comparison of constant -128 with expression of type 'unsigned char' is always false [-Werror,-Wtautological-constant-out-of-range-compare]
  if ((unsigned char)last_element->first[0] < BI_PREFIX_CHAR) {
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/cls/rgw/cls_rgw.cc:221:46: error: result of comparison of constant -128 with expression of type 'unsigned char' is always true [-Werror,-Wtautological-constant-out-of-range-compare]
  if ((unsigned char)first_element->first[0] > BI_PREFIX_CHAR) {
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~
3 errors generated.
```

Reply:
then you should remove the C-style conversions used there (they were probably needed because of the original problem).

Reply:
@yuvalif done. can you check please?

Reply:
ok. i see 0x80 = 128 which does not fit as a positive char.
so, keep as "unsigned char" and do the casting.
* probably better with `static_cast<unsigned char>` instead of the C-style
* i thin kwe should do the explicit casting the other direction as well (from `unsigned char` to the `std::string` that expects `char`)